src/limbo/model/gp.hpp

```diff
@@ -117,11 +119,14 @@ namespace limbo {
             void add_sample(const Eigen::VectorXd& sample, const Eigen::VectorXd& observation, double noise)
             {
                 if (_samples.empty()) {
+		  if (_dim_in != sample.size()) {
                     _dim_in = sample.size();
                     _kernel_function = KernelFunction(_dim_in); // the cost of building a functor should be relatively low
-
-                    _dim_out = observation.size();
-                    _mean_function = MeanFunction(_dim_out); // the cost of building a functor should be relatively low
+		  }
+		  if (_dim_out != observation.size()) {
```

Comment:
I agree with what you said, but you can also define the dim_in and dim_out in the GP constructor and it may happen that the dims that you specified in the constructor are not the good ones (for instance when you don't know the dim of your observations when you construct your GP).

Moreover, here you reconstruct the Mean (and Kernel) objects after adding the first sample. this means that in my case, I would have to add the "alteration" of the Mean object between to "add_sample" instead of just after the construction of my GP.